pymeasure/display/windows.py

```diff
@@ -172,6 +173,19 @@ def __init__(self, procedure_class, inputs=(), displays=(), x_axis=None, y_axis=
         log.setLevel(log_level)
         self.log.setLevel(log_level)
         self.x_axis, self.y_axis = x_axis, y_axis
+
+        # Check if the get_estimates function is reimplemented
+        proc = self.procedure_class()
+        try:
+            proc.get_estimates()
+        except NotImplementedError:
+            self.use_estimator = False
+        except TypeError:
+            # Raised if arguments are asked by the get_estimates method.
+            self.use_estimator = True
+        else:
+            self.use_estimator = True
+
         self._setup_ui()
```

Comment:
An alternative implementation could be:
```
        self.use_estimator = True
        try:
            proc.get_estimates()
        except NotImplementedError:
            self.use_estimator = False
```
It is more compact, but I am not sure whether it is better.

Reply:
That should also work (though the TypeError should then also be caught with as pass), but I now have another approach by comparing the `get_estimates` methods:

> self.use_estimator = not self.procedure_class.get_estimates == Procedure.get_estimates

This way, there is no need to really instantiate the procedure class

Reply:
Much better than my proposal 😉